lib/ImplicitDiscreteSolve/src/controller.jl

```diff
@@ -0,0 +1,52 @@
+Base.@kwdef struct KantorovichTypeController <: OrdinaryDiffEqCore.AbstractController
+    Θmin::Float64
+    p::Int64
+    Θreject::Float64 = 0.95
+    Θbar::Float64 = 0.5
+    γ::Float64    = 0.95
+    qmin::Float64 = 1/5
+    qmax::Float64 = 5.0
+end
+
+function OrdinaryDiffEqCore.default_controller(alg::IDSolve, cache::IDSolveCache, _1, _2, _3)
+    return KantorovichTypeController(;Θmin=1//8, p=1)
+end
+
+function OrdinaryDiffEqCore.stepsize_controller!(
+    integrator, controller::KantorovichTypeController, alg::IDSolve
+)
+    @inline g(x) = √(1+4x) - 1
+
+    # Adapt dt with a priori estimate (Eq. 5.24)
+    (; Θks) = integrator.cache
+    (; Θbar, γ, Θmin, qmin, qmax, p) = controller
+
+    Θ₀ = length(Θks) > 0 ? max(first(Θks), Θmin) : Θmin
+    q = clamp(γ * (g(Θbar)/(g(Θ₀)))^(1/p), qmin, qmax)
+
+    return q
```

Comment:
I think I am misunderstanding what exactly should be returned here and how it integrates with the dt update logic. If I want to scale the time step by a factor of 2, then what exactly should be returned here?

Reply:
2